### PR TITLE
Avalanche fixes

### LINKED ISSUE #2568: Add Native Avalanche Support to Tally Ho
### Requirements
- [ ] Send / Receive Assets
- [ ] Swaps
- [ ] NFT Support via SimpleHash
- [ ] Gas Price Preview
- [ ] Dapp connection / tx signing
- [ ] Gas Price Adjust
- [ ] If #2547 or similar has been merged - preference for a generic rpc provider.
- [ ] Assets showing up in Overview
- [ ] ENS Name Resolution
- [ ] Ensure support for popular dapps  (cc @mr-michael @alsoknownaslj @Naxsun  to provide list in a separate issue).

### Stretch Goal
- [ ] Name resolution via [avvy.domains](https://docs.avvy.domains/)

### Useful References
- https://github.com/tallycash/extension/pull/2287/files
- https://github.com/tallycash/extension/pull/1347/files

### Notes
- I expect we'll want to open multiple PRs to ultimately close out this ticket - trying to tackle this in one PR might result in a large amount of code to review.
- PRs and issues open for avalanche-specific bugs encountered while implementing avalanche support should be added to the [`Avalanche Support`](https://github.com/tallycash/extension/milestone/14) milestone.

### PR DESCRIPTION
**Pull request by hyphenized**
Closes #2568

## To Test:
- [x] Send / Receive Assets
- [x] Swaps
- [x] Check NFT Support
- [x] Gas Price Previews
- [x] dApp connection / Tx signing
- [x] Ledger Support
- [x] Gas Price Modification
- [x] Assets showing up in Overview
- [x] ENS Name Resolution
- [x] Test interactions with dApps

Latest build: [extension-builds-2701](https://github.com/tallycash/extension/suites/9620018608/artifacts/459316682) (as of Thu, 01 Dec 2022 21:58:31 GMT).

### CODE REVIEW

File: ui/_locales/en/messages.json [hunk 1]
```diff
@@ -342,6 +342,7 @@
     "testnet": "Test Network",
     "l2": "L2 scaling solution",
     "compatibleChain": "Ethereum-compatible blockchain",
+    "avalanche": "Mainnet C-Chain",
```
Comment (0xDaedalus) [inline ui/_locales/en/messages.json:345]:
```suggestion
    "avalanche": "Mainnet C-Chain (beta)",
```

**Comment by 0xDaedalus** [comment]
Things are looking fantastic here!

Only thing not working is gas price modification but I believe this is not an avalanche-specific issue so lets not let it block us.

Dapps tested:
- AAVE
- BenQi
- TraderJoe
- Stargate
- GMX

File: window-provider/wallet-connection-handlers.ts [hunk 2]
```diff
@@ -60,6 +60,36 @@ const findAndReplaceUniswapInjectedOption = (): void => {
   }
 }
 
+const findAndReplaceJoeMetamaskOption = (): void => {
+  let maybeButton = document.getElementById("connect-INJECTED")
+
+  if (!maybeButton) {
+    document
+      .querySelectorAll<HTMLButtonElement>("[aria-modal] button")
```
[the remark this answers is not in the document]
Comment (hyphenized) [reply window-provider/wallet-connection-handlers.ts]: Thanks, You're right. There's no need to iterate the entire DOM, I've changed the injection code to actually use the mutation observer dispatched changes

File: window-provider/index.ts [hunk 3]
```diff
@@ -72,6 +75,15 @@ export default class TallyWindowProvider extends EventEmitter {
     }
   >()
 
+  // TODO: These are used by MetaMask and accessed by some dApps
```
[the remark this answers is not in the document]
Comment (hyphenized) [reply window-provider/index.ts]: I've set a host check with the conflicting dApp ([core.app](https://core.app)) to conditionally apply this state mock. This is an internal property of the base MM provider which should not be accessed by dApps. However, given this case, I think if we're impersonating MetaMask we should consider implementing these APIs 😬

**Review by 0xDaedalus — APPROVED** [review]
avalanche

File: window-provider/wallet-connection-handlers.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (greg-nagy) [reply window-provider/wallet-connection-handlers.ts]: This change breaks traderjoe for me — no metamask, no tally button

CleanShot 2022-12-05 at 13 32 35

File: window-provider/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (greg-nagy) [reply window-provider/index.ts]: I feel like we are batman at the bottom of the pit ...

File: window-provider/wallet-connection-handlers.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (greg-nagy) [reply window-provider/wallet-connection-handlers.ts]: The default wallet is a good catch — I missed that.
 
But the primary source of my confusion is that we remove metamask button if tally is installed but not set to default.
And we display tally logo if I have both MM and tally, but tally is not set to be default. And MM opens if I press the button. 

Is this traderjoe specific or all our injections behave the same way?

**Pull request merged**